src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceAuthoriseUserTaskTest.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    @Test
+    void should_get_a_task_to_user_with_divorce_only_permissions() {
+        final String taskId = "8d6cc5cf-c973-11eb-bdba-0242ac111018";
+        final String caseId = "1623278362431018";
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+        final Map<String, String> tcAttributes = Map.of(
+            RoleAttributeDefinition.CASE_TYPE.value(), "Asylum",
+            RoleAttributeDefinition.JURISDICTION.value(), "IA",
+            RoleAttributeDefinition.CASE_ID.value(), caseId
+        );
+        RoleAssignment roleAssignment = RoleAssignment.builder().roleName("tribunal-caseworker")
+            .classification(Classification.PUBLIC)
+            .beginTime(LocalDateTime.now().minusYears(1))
+            .endTime(LocalDateTime.now().plusYears(1))
+            .authorisations(List.of("DIVORCE"))
+            .grantType(GrantType.CHALLENGED)
+            .attributes(tcAttributes)
+            .build();
+        roleAssignments.add(roleAssignment);
+
+        AccessControlResponse accessControlResponse = new AccessControlResponse(null, roleAssignments);
+        permissionsRequired.add(PermissionTypes.READ);
+
+        final Optional<TaskResource> task = cftQueryService.getTask(taskId, accessControlResponse, permissionsRequired);
+        Assertions.assertThat(task.isPresent()).isTrue();
+        Assertions.assertThat(task.get().getTaskId()).isEqualTo(taskId);
+        Assertions.assertThat(task.get().getCaseId()).isEqualTo(caseId);
+    }
+
+    @Test
+    void should_get_a_task_to_user_with_unknown_permissions() {
+        final String taskId = "8d6cc5cf-c973-11eb-bdba-0242ac111018";
+        final String caseId = "1623278362431018";
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+        final Map<String, String> tcAttributes = Map.of(
+            RoleAttributeDefinition.CASE_TYPE.value(), "Asylum",
+            RoleAttributeDefinition.JURISDICTION.value(), "IA",
+            RoleAttributeDefinition.CASE_ID.value(), caseId
+        );
+        RoleAssignment roleAssignment = RoleAssignment.builder().roleName("tribunal-caseworker")
+            .classification(Classification.PUBLIC)
+            .beginTime(LocalDateTime.now().minusYears(1))
+            .endTime(LocalDateTime.now().plusYears(1))
+            .authorisations(List.of("NO Role"))
+            .grantType(GrantType.CHALLENGED)
+            .attributes(tcAttributes)
+            .build();
+        roleAssignments.add(roleAssignment);
+
+        AccessControlResponse accessControlResponse = new AccessControlResponse(null, roleAssignments);
+        permissionsRequired.add(PermissionTypes.READ);
+
+        final Optional<TaskResource> task = cftQueryService.getTask(taskId, accessControlResponse, permissionsRequired);
+        Assertions.assertThat(task.isPresent()).isFalse();
+    }
+
+    @Test
+    void should_get_a_task_to_user_with_no_permissions() {
```

Comment:
can we rename this to be indicative of what the test is checking? for example should_not_get